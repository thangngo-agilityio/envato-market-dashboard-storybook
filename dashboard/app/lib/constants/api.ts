// TODO: Will provide latter
export const BASE_API = '';
export const USERS_API = process.env.NEXT_PUBLIC_API_USER || BASE_API;
export const STATISTICAL_API =
  process.env.NEXT_PUBLIC_STATISTICAL_API || BASE_API;

export const END_POINTS = {
  AUTHEN_USERS: '/authenUsers',
  USERS: '/users',
  TRANSACTIONS: '/transactions',
  PRODUCTS: '/products',
  RECENT_ACTIVITIES: '/activities',
  STATISTICS: '/statistics',
  REVENUE: '/revenue',
  EFFICIENCY: '/efficiency',
  NOTIFICATION: '/notifications',
  EMPLOYEES: '/employees',
  OVERALL_BALANCE: '/overall-balance',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  DELETE_TRANSACTION: '/delete-transactions',
  EDIT_TRANSACTION: '/edit-transactions',
  CREATE_PIN: '/create-pin-code',
  CONFIRM_PIN: '/pin-code',
  UPDATE_PASSWORD: '/update-password',
  MY_WALLET: '/wallets',
  ADD_MONEY: '/add-money',
  SEND_MONEY: '/send-money',
  SUPPORT: '/supports',
  LOCK: '/lock',
  UNLOCK: '/unlock',
  ADMIN: '/admin',
  SETTINGS: '/settings',
  LOGIN: '/login',
  WALLET: '/my-wallets',
};

export const SEARCH_PARAM = {
  EMAIL: 'email',
  PASSWORD: 'password',
  TRANSACTION_NAME: 'transactionName',
};
