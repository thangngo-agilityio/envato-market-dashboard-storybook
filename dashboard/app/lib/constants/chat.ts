export const FIREBASE_CHAT = {
  USERS: 'users',
  USER_CHATS: 'userChats',
  CHATS: 'chats',
};

export const USER_CHATS_FIELD = {
  USER_INFO: '.userInfo',
  DATE: '.date',
  LAST_MESSAGE: '.lastMessage',
};
