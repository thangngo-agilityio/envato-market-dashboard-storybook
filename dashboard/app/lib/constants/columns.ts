// Types
import {
  TDataSource,
  TProduct,
  TProductResponse,
  TRecentActivities,
  TTransaction,
} from '../interfaces';

export const COLUMNS_DASHBOARD = (
  onRenderHead: (title: string, key: string) => void,
  onRenderRole: (role: TTransaction) => void,
  onRenderBody: ({ id, image, name }: TDataSource) => void,
  onRenderActionIcon: (data: TTransaction) => void,
  onRenderSpent: (spent: TTransaction) => void,
  onRenderEmail: (email: TTransaction) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
    renderBody: onRenderEmail,
  },
  {
    title: 'Location',
    key: 'location',
    renderHead: onRenderHead,
  },
  {
    title: 'Spent',
    key: 'spent',
    renderHead: onRenderHead,
    renderBody: onRenderSpent,
  },
  {
    title: 'Role',
    key: 'role',
    renderHead: onRenderHead,
    renderBody: onRenderRole,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_HISTORY = (
  renderHead: (title: string, key: string) => void,
  renderBody: ({ id, image, name }: TDataSource) => void,
  renderPaymentStatus: ({ paymentStatus }: TDataSource) => void,
  renderTransactionStatus: ({ paymentStatus }: TDataSource) => void,
  renderSpent: (spent: TTransaction) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: renderHead,
    renderBody: renderBody,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: renderHead,
  },
  {
    title: 'Amount',
    key: 'spent',
    renderHead: renderHead,
    renderBody: renderSpent,
  },
  {
    title: 'Payment',
    key: 'paymentStatus',
    renderHead: renderHead,
    renderBody: renderPaymentStatus,
  },
  {
    title: 'Status',
    key: 'transactionStatus',
    renderHead: renderHead,
    renderBody: renderTransactionStatus,
  },
];

export const COLUMNS_PRODUCTS = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, name }: TDataSource) => void,
  onRenderGallery: ({ imageURLs, name }: TProduct) => void,
  onRenderPrice: (amount: TProduct) => void,
  onRenderStatus: (productStatus: TDataSource) => void,
  onRenderQuantity: (stock: TProduct) => void,
  onRenderActionIcon: (data: TProductResponse) => void,
) => [
  {
    title: 'Product Name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Gallery Thumbnail',
    key: 'imageURLs',
    renderHead: onRenderHead,
    renderBody: onRenderGallery,
  },
  {
    title: 'Price',
    key: 'price',
    renderHead: onRenderHead,
    renderBody: onRenderPrice,
  },
  {
    title: 'Status',
    key: 'productStatus',
    renderHead: onRenderHead,
    renderBody: onRenderStatus,
  },
  {
    title: 'Quantity',
    key: 'quantity',
    renderHead: onRenderHead,
    renderBody: onRenderQuantity,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: onRenderHead,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_RECENT_ACTIVITIES = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, actionName }: TDataSource) => void,
  onRenderName: (actionName: TRecentActivities) => void,
  onRenderEmail: (email: TRecentActivities) => void,
) => [
  {
    title: 'ID Action',
    key: '_id',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Action Name',
    key: 'actionName',
    renderHead: onRenderHead,
    renderBody: onRenderName,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
    renderBody: onRenderEmail,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: onRenderHead,
  },
];
