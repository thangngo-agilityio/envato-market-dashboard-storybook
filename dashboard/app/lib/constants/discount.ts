export const DEFAULT_DISCOUNT_PERCENTAGE = 0.6;
export const LIMIT_OF_BONUS = 5;
