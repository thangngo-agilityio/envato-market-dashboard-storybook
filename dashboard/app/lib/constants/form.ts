import { ERROR_MESSAGES } from './messages';
import { REGEX } from './regex';

export const AUTH_SCHEMA = {
  FIRST_NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('First Name'),
  },
  LAST_NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Last Name'),
  },
  STREET: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Street'),
  },
  STATE: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('State'),
  },
  CITY: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('City'),
  },
  EMAIL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Email'),
    pattern: {
      value: REGEX.EMAIL,
      message: ERROR_MESSAGES.EMAIL_INVALID,
    },
  },
  PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Password'),
  },
  CONFIRM_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Confirm password'),
    validate: (val: string, { password }: { password: string }) => {
      if (password && val !== password) {
        return ERROR_MESSAGES.PASSWORD_NOT_MATCH;
      }
    },
  },
  REMEMBER_ME: {
    required: false,
  },
  AGREE_POLICY: {
    validate: (
      value: boolean,
      //eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unused-vars
      { isAcceptPrivacyPolicy: __, ...fieldValues }: any,
    ) => Object.values(fieldValues).every((value) => value) && value,
  },

  PHONE_NUMBER: {
    pattern: {
      value: REGEX.PHONE_NUMBER,
      message: ERROR_MESSAGES.PHONE_NUMBER_INVALID,
    },
  },

  AVATAR_URL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Avatar'),
    pattern: {
      value: REGEX.IMG,
      message: ERROR_MESSAGES.UPLOAD_IMAGE,
    },
  },

  FACEBOOK: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.FACEBOOK_INVALID,
    },
  },

  TWITTER: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.TWITTER_INVALID,
    },
  },

  LINKEDIN: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.LINKEDIN_INVALID,
    },
  },

  YOUTUBE: {
    pattern: {
      value: REGEX.URL,
      message: ERROR_MESSAGES.YOUTUBE_INVALID,
    },
  },

  ZIP_CODE: {
    pattern: {
      value: REGEX.ZIP_CODE,
      message: ERROR_MESSAGES.ZIP_CODE_INVALID,
    },
  },

  PIN_CODE: {
    required: true,
    validate: (value: string) => value.length === 4,
  },

  TRANSFER_AMOUNT: {
    required: true,
    validate: (value: string) => !!value.length && !isNaN(+value),
  },
  OLD_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Old password'),
  },
  NEW_PASSWORD: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('New password'),
    validate: (value: string) => {
      if (!REGEX.LENGTH_IS_EIGHT.test(value)) {
        return ERROR_MESSAGES.PASS_WORD_SHORT;
      }

      if (!REGEX.PASSWORD.test(value)) {
        return ERROR_MESSAGES.PASS_WORD_WEAK;
      }

      return true;
    },
  },

  NAME: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Name'),
  },

  AMOUNT: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Amount'),
    validate: (value: string) => {
      if (Number(value) <= 0) {
        return ERROR_MESSAGES.AMOUNT_INVALID;
      }
      if (!REGEX.LIMIT_NUMBER.test(value)) {
        return ERROR_MESSAGES.LIMIT_AMOUNT;
      }

      return true;
    },
  },

  QUANTITY: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Quantity'),
    validate: (value: string) => {
      if (Number(value) < 0) {
        return ERROR_MESSAGES.QUANTITY_INVALID;
      }
      if (!REGEX.LIMIT_NUMBER.test(value)) {
        return ERROR_MESSAGES.LIMIT_QUANTITY;
      }

      return true;
    },
  },

  GALLERY_THUMBNAIL: {
    required: ERROR_MESSAGES.FIELD_REQUIRED('Gallery Thumbnail'),
  },
};
