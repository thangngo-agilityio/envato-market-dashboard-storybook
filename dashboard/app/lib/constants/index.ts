export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
export * from './faq';
export * from './avatar';
export * from './sizes';
export * from './setting';
export * from './termAndCondition';
export * from './notification';
export * from './authentication';
export * from './discount';
export * from './metadata';
export * from './og-properties';
export * from './chat';
export * from './product';
