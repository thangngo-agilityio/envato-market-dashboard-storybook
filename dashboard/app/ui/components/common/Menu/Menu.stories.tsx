import type { Meta, StoryObj } from '@storybook/react';
import { Box } from '@chakra-ui/react';

// components
import { Menu } from '@/ui/components';
import { HELP_ITEM_LIST, OTHER_ITEM_LIST } from '@/lib/constants';

const meta: Meta<typeof Menu> = {
  title: 'Custom Components/Menu',
  tags: ['autodocs'],
  component: Menu,
  argTypes: {
    title: {
      description: 'The title of the Menu component',
    },

    listItem: {
      description: 'The list item of the menu',
    },

    isExpandSidebar: {
      description: 'Determine the mode of the menu in sidebar: expand or mini',
    },
  },
  decorators: [
    (Story) => (
      <Box bg="background.component.primary">
        <Story />
      </Box>
    ),
  ],
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof Menu>;

export const Expand: Story = {
  args: {
    listItem: OTHER_ITEM_LIST,
    title: 'Expand Menu Component',
  },
};

export const Minify: Story = {
  args: {
    title: 'Minify Menu Component',
    listItem: HELP_ITEM_LIST,
    isExpandSidebar: true,
  },
};
