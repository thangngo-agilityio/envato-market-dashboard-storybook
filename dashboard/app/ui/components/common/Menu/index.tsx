'use client';

import {
  Fragment,
  MouseEvent,
  ReactElement,
  ReactNode,
  useCallback,
} from 'react';
import { Flex, Heading, List, ListItem, Text, VStack } from '@chakra-ui/react';

// Components
import { Navigation } from '@/ui/components';

// Constants
import { ROUTES } from '@/lib/constants';

export type TMenuItem = {
  id: number;
  leftIcon?: () => ReactElement;
  rightIcon?: ReactNode;
  menuItemContent?: string;
  destination: string;
};

export type MenuProps = {
  title?: string;
  listItem: Array<TMenuItem>;
  isExpandSidebar?: boolean;
  onClickMenuItem?: () => void;
  onSignOut?: () => void;
};

const Menu = ({
  title = '',
  listItem,
  isExpandSidebar = false,
  onSignOut,
  onClickMenuItem,
}: MenuProps) => {
  const handleSignOut = useCallback(
    (e: MouseEvent<HTMLAnchorElement>) => {
      e.preventDefault();
      onSignOut && onSignOut();
    },
    [onSignOut],
  );

  return (
    <VStack mb={isExpandSidebar ? 9 : 5} w="full" overscroll="full">
      {!isExpandSidebar && (
        <Heading
          as="h4"
          borderColor="border.primary"
          borderBottomWidth="1px"
          w="full"
          lineHeight={1.75}
          fontSize="sm"
          color="text.secondary"
          _light={{
            borderColor: 'gray.100',
            color: 'gray.600',
          }}
          fontWeight="md"
        >
          {title}
        </Heading>
      )}

      <List
        mt={2.5}
        aria-label="list-icon"
        spacing={isExpandSidebar ? 5.5 : 2.75}
        w="full"
      >
        {listItem.map(
          ({ leftIcon, rightIcon, destination, menuItemContent, id }) => {
            if (!menuItemContent) return <></>;
            const LeftIconComponent = leftIcon || Fragment;

            const handleClick =
              destination === `/${ROUTES.SIGN_OUT}`
                ? handleSignOut
                : onClickMenuItem;

            return (
              <ListItem key={id} aria-label="item-icon">
                {isExpandSidebar ? (
                  <Navigation destination={destination} onClick={handleClick}>
                    <LeftIconComponent />
                  </Navigation>
                ) : (
                  <Navigation destination={destination} onClick={handleClick}>
                    <Flex alignItems="center" justifyContent="space-between">
                      <Flex
                        gap={2.5}
                        alignItems="center"
                        justifyContent="space-between"
                      >
                        <LeftIconComponent />

                        <Text
                          fontSize="lg"
                          fontWeight="md"
                          color="text.primary"
                          transition="all .25s ease-in-out"
                        >
                          {menuItemContent}
                        </Text>
                      </Flex>

                      {rightIcon}
                    </Flex>
                  </Navigation>
                )}
              </ListItem>
            );
          },
        )}
      </List>
    </VStack>
  );
};

export default Menu;
