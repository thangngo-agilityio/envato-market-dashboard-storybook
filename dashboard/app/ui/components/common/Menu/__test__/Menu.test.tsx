// component
import { HELP_ITEM_LIST, OTHER_ITEM_LIST, ROUTES } from '@/lib/constants';
import Menu from '..';
import userEvent from '@testing-library/user-event';
import * as navigation from 'next/navigation';

describe('Menu component', () => {
  test('renders Logo with content', () => {
    const { container } = testLibReactUtils.render(
      <Menu listItem={HELP_ITEM_LIST} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('handle logout', async () => {
    const routerPushMock = jest.fn();
    jest.spyOn(navigation, 'useRouter').mockReturnValue({
      push: routerPushMock,
    } as unknown as ReturnType<typeof navigation.useRouter>);
    const { getByText } = testLibReactUtils.render(
      <Menu listItem={OTHER_ITEM_LIST} />,
    );

    const signOut = getByText(/sign out/i);

    await userEvent.click(signOut);
    testLibReactUtils.waitFor(() => {
      expect(routerPushMock).toHaveBeenCalledWith(`/${ROUTES.SIGN_OUT}`);
    });
  });
});
