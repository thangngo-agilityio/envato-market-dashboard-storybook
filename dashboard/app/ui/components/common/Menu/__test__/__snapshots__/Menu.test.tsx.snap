// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders Logo with content 1`] = `
<div>
  <div
    class="chakra-stack css-od3pg5"
  >
    <h4
      class="chakra-heading css-7ikyn6"
    />
    <ul
      aria-label="list-icon"
      class="css-1gvjgvu"
      role="list"
    >
      <li
        aria-label="item-icon"
        class="css-0"
      >
        <a
          aria-label="navigate-item"
          class="chakra-link css-fx4yts"
          href="/supports"
        >
          <div
            class="css-1lekzkb"
          >
            <div
              class="css-156rdqq"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 20 18"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="path-1"
                  d="M5 2V11C5 12.1046 5.89543 13 7 13H18C19.1046 13 20 12.1046 20 11V2C20 0.895431 19.1046 0 18 0H7C5.89543 0 5 0.89543 5 2Z"
                  fill="#1A202C"
                />
                <path
                  class="path-2"
                  d="M0 15C0 13.8954 0.895431 13 2 13H2.17157C2.70201 13 3.21071 13.2107 3.58579 13.5858C4.36683 14.3668 5.63317 14.3668 6.41421 13.5858C6.78929 13.2107 7.29799 13 7.82843 13H8C9.10457 13 10 13.8954 10 15V16C10 17.1046 9.10457 18 8 18H2C0.89543 18 0 17.1046 0 16V15Z"
                  fill="#22C55E"
                />
                <path
                  class="path-2"
                  d="M7.5 9.5C7.5 10.8807 6.38071 12 5 12C3.61929 12 2.5 10.8807 2.5 9.5C2.5 8.11929 3.61929 7 5 7C6.38071 7 7.5 8.11929 7.5 9.5Z"
                  fill="#22C55E"
                />
                <path
                  class="path-2"
                  clip-rule="evenodd"
                  d="M8.25 4.5C8.25 4.08579 8.58579 3.75 9 3.75L16 3.75C16.4142 3.75 16.75 4.08579 16.75 4.5C16.75 4.91421 16.4142 5.25 16 5.25L9 5.25C8.58579 5.25 8.25 4.91421 8.25 4.5Z"
                  fill="#22C55E"
                  fill-rule="evenodd"
                />
                <path
                  class="path-2"
                  clip-rule="evenodd"
                  d="M11.25 8.5C11.25 8.08579 11.5858 7.75 12 7.75L16 7.75C16.4142 7.75 16.75 8.08579 16.75 8.5C16.75 8.91421 16.4142 9.25 16 9.25L12 9.25C11.5858 9.25 11.25 8.91421 11.25 8.5Z"
                  fill="#22C55E"
                  fill-rule="evenodd"
                />
              </svg>
              <p
                class="chakra-text css-1bcsbvl"
              >
                Supports
              </p>
            </div>
          </div>
        </a>
      </li>
      <li
        aria-label="item-icon"
        class="css-0"
      >
        <a
          aria-label="navigate-item"
          class="chakra-link css-fx4yts"
          href="/settings"
        >
          <div
            class="css-1lekzkb"
          >
            <div
              class="css-156rdqq"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="path-1"
                  d="M13.0606 2H10.9394C9.76787 2 8.81817 2.89543 8.81817 4C8.81817 5.26401 7.46574 6.06763 6.35556 5.4633L6.24279 5.40192C5.22823 4.84963 3.93091 5.17738 3.34515 6.13397L2.28455 7.86602C1.69879 8.8226 2.0464 10.0458 3.06097 10.5981C4.17168 11.2027 4.17168 12.7973 3.06096 13.4019C2.0464 13.9542 1.69879 15.1774 2.28454 16.134L3.34515 17.866C3.93091 18.8226 5.22823 19.1504 6.24279 18.5981L6.35555 18.5367C7.46574 17.9324 8.81817 18.736 8.81817 20C8.81817 21.1046 9.76787 22 10.9394 22H13.0606C14.2321 22 15.1818 21.1046 15.1818 20C15.1818 18.736 16.5343 17.9324 17.6445 18.5367L17.7572 18.5981C18.7718 19.1504 20.0691 18.8226 20.6548 17.866L21.7155 16.134C22.3012 15.1774 21.9536 13.9542 20.939 13.4019C19.8283 12.7973 19.8283 11.2027 20.939 10.5981C21.9536 10.0458 22.3012 8.82262 21.7155 7.86603L20.6548 6.13398C20.0691 5.1774 18.7718 4.84965 17.7572 5.40193L17.6445 5.46331C16.5343 6.06765 15.1818 5.26402 15.1818 4C15.1818 2.89543 14.2321 2 13.0606 2Z"
                  fill="#1A202C"
                />
                <path
                  class="path-2"
                  d="M15.75 12C15.75 14.0711 14.0711 15.75 12 15.75C9.92893 15.75 8.25 14.0711 8.25 12C8.25 9.92893 9.92893 8.25 12 8.25C14.0711 8.25 15.75 9.92893 15.75 12Z"
                  fill="#22C55E"
                />
              </svg>
              <p
                class="chakra-text css-1bcsbvl"
              >
                Settings
              </p>
            </div>
          </div>
        </a>
      </li>
    </ul>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
